provider/core/src/serde/ser.rs

```diff
@@ -58,18 +49,19 @@ impl DataPayload<SerializeMarker> {
     ///
     /// // Serialize the payload to a JSON string
     /// let mut buffer: Vec<u8> = vec![];
-    /// payload.into_serializable().serialize(
-    ///     &mut <dyn erased_serde::Serializer>::erase(
-    ///         &mut serde_json::Serializer::new(&mut buffer)
-    ///     )
-    /// ).expect("Serialization should succeed");
+    /// payload.into_serializable().serialize(&mut serde_json::Serializer::new(&mut buffer))
+    ///     .expect("Serialization should succeed");
     /// assert_eq!("{\"message\":\"(und) Hello World\"}".as_bytes(), buffer);
     /// ```
-    pub fn serialize(
-        &self,
-        mut serializer: &mut dyn erased_serde::Serializer,
-    ) -> Result<(), Error> {
-        self.get().erased_serialize(&mut serializer)?;
+    pub fn serialize<S>(&self, serializer: S) -> Result<(), DataError>
```

Comment:
Praise: Good idea bringing the erased_serde interface down one more layer, so that child crates don't need the dependency